runtime/bindings/python/vm.h

```diff
@@ -50,44 +57,33 @@ struct ApiPtrAdapter<iree_vm_invocation_t> {
   }
 };
 
+template <>
+struct ApiPtrAdapter<iree_vm_ref_t> {
+  static void Retain(iree_vm_ref_t* b) {
+    iree_vm_ref_t out_ref;
+    out_ref.ptr = nullptr;
```

Comment:
you'll want to fully zero init for safety - the type being 0 is important as some places are doing .type comparisons (isa/etc)

Reply:
(we should probably have an `iree_vm_ref_empty()` helper or something to avoid the need to memset - I can add that later on if you just want to `{0}` init or or memset for now)